src/event/target.js

```diff
@@ -44,6 +44,21 @@ class EventTarget {
     this.listeners[type] = reject(arrayOfListeners, listener => listener === removingListener);
   }
 
+  /*
+   * Checks if a given event type has registered event listeners
+   *
+   * @param {string} type - the type of event (ie: 'open', 'message', etc.)
+   */
+  hasEventListeners(type) {
```

Comment:
please move this logic to `hasListeners` method. Native EventTarget doesn't have it